bndtools.core/src/bndtools/explorer/BndtoolsExplorer.java

```diff
@@ -34,9 +45,34 @@
 
 	private final FilterPanelPart	filterPart	= new FilterPanelPart(Plugin.getDefault()
 		.getScheduler());
-	private PropertyChangeListener	listener;
+	private PropertyChangeListener	filterListener;
 	private Glob					glob;
 	boolean							installed;
+	private IWorkbench				workbench	= PlatformUI.getWorkbench();
+	private IProject				selectedProject;
+	private IPartListener			partListener	= new IPartListener() {
+
+														@Override
+														public void partOpened(IWorkbenchPart part) {}
+
+														@Override
+														public void partDeactivated(IWorkbenchPart part) {}
+
+														@Override
+														public void partClosed(IWorkbenchPart part) {}
+
+														@Override
+														public void partBroughtToTop(IWorkbenchPart part) {}
+
+														@Override
+														public void partActivated(IWorkbenchPart part) {
+															if (part instanceof IEditorPart) {
```

Comment:
I'm not sure I understand the expected behavior.  It seems the "link with editor" toggle button that is already in the toolbar does this behavior already.  I don't see this new PartListener you added as adding much value unless I'm missing something.

Reply:
No, it seems to not do this. I think what is happening is that the link tries to reveal the object, but if that object is not in the filtered list, it ignores the reveal. I.e. it only reveals objects that are allowed to be displayed. This listener resets the filter selection before the object is revealed.  It seems to follow the Link button. I.e. when it is off, the project is not shown in the list

Reply:
That is the missing piece.  You need to have an active filter in place, and then open an editor that is in a project that is not currently being shown in the filtered list.

Reply:
Actually, I just tried that, but I'm still not seeing your original problem.  Maybe we should jump on a call so I can see what you are seeing.